rclpy/rclpy/node.py

```diff
@@ -1652,3 +1652,68 @@ def assert_liveliness(self) -> None:
         """
         with self.handle as capsule:
             _rclpy.rclpy_assert_liveliness(capsule)
+
+    def _get_info_by_topic(
+        self,
+        topic_name: str,
+        no_mangle: bool,
+        func: Callable[[object, str, bool], List[Dict]]
+    ) -> List[Dict]:
+        fq_topic_name = expand_topic_name(topic_name, self.get_name(), self.get_namespace())
+        validate_full_topic_name(fq_topic_name)
+        with self.handle as node_capsule:
+            return func(node_capsule, fq_topic_name, no_mangle)
+
+    def get_publishers_info_by_topic(
+            self, topic_name: str, no_mangle: bool = False) -> List[Dict]:
+        """
+        Return a list of publishers publishing to a given topic.
+
+        The returned parameter is a list of dictionaries, where each dictionary will contain
+        the node name, node namespace, topic type, topic endpoint's GID, and its QoS profile.
```

Comment:
Would it be better to have a class for this rather just a dict? The reason I ask is that it would be nice to have a class with a `__str__()` method so the printing logic could be centralized. Right now you have to touch a lot of places of code if we add something to this info dict.

Reply:
I think that that improvement makes sense.

Reply:
I also agree. I will need to spend some time to implement that.

Reply:
Thanks, hopefully it's not too much work.

Reply:
Done.